src/qibolab/_core/instruments/qcs.py

```diff
@@ -0,0 +1,343 @@
+                            phase=sweep_param_map.get(
+                                "relative_phase", pulse.probe.relative_phase
+                            ),
+                        )
+                        integration_filter = qcs.IntegrationFilter(qcs_pulse)
+                        program.add_waveform(qcs_pulse, probe_virtual_channel)
+                        program.add_acquisition(integration_filter, virtual_channel)
+
+            elif isinstance(channel, IqChannel):
+
+                pulses = []
+                if channel_id in sweeper_channel_map:
+                    frequency = sweeper_channel_map[channel_id]
+                else:
+                    frequency = configs[channel_id].frequency
+
+                for pulse in sequence.channel(channel_id):
+                    sweep_param_map = sweeper_pulse_map.get(pulse.id, {})
+
+                    if pulse.kind == "delay":
+                        qcs_pulse = qcs.Delay(
+                            sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            )
+                        )
+                    elif pulse.kind == "virtualz":
+                        qcs_pulse = qcs.PhaseIncrement(
+                            phase=sweep_param_map.get("relative_phase", pulse.phase)
+                        )
+                    elif pulse.kind == "pulse":
+                        qcs_pulse = generate_qcs_rfwaveform(
+                            duration=sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            ),
+                            envelope=pulse.envelope,
+                            amplitude=sweep_param_map.get("amplitude", pulse.amplitude),
+                            frequency=frequency,
+                            phase=sweep_param_map.get(
+                                "relative_phase", pulse.relative_phase
+                            ),
+                        )
+                        if pulse.envelope.kind == "drag":
+                            qcs_pulse = qcs_pulse.drag(coeff=pulse.envelope.beta)
+                    else:
+                        raise ValueError("Unrecognized pulse type", pulse.kind)
+
+                    pulses.append(qcs_pulse)
+
+                program.add_waveform(pulses, virtual_channel)
+
+            elif isinstance(channel, DcChannel):
+
+                pulses = []
+                sweep_param_map = sweeper_pulse_map.get(pulse.id, {})
+
+                for pulse in sequence.channel(channel_id):
+                    if pulse.kind == "delay":
+                        duration = (
+                            sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            ),
+                        )
+                    elif pulse.kind == "pulse":
+                        qcs_pulse = qcs.DCWaveform(
```

Comment:
Even this `qcs_pulse` is unused, since the `DcChannel` branch is missing a call like
https://github.com/qiboteam/qibolab/blob/187034350ab4c0a16ec3de09c5e3e00c6d15e414/src/qibolab/_core/instruments/qcs.py#L245
in the `IqChannel` one.

Reply:
Fixed in https://github.com/qiboteam/qibolab/pull/944/commits/5295cd35cd5ed4aa75a44bd4247863eb51fd7eb6